src/mappings/DAOTracker/mapping.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  BlacklistDAO,
+  DAOTracker,
+  ResetDAO,
+  TrackDAO,
+} from '../../types/DAOTracker/DAOTracker';
+import {
+  AvatarContract,
+  BlacklistedDAO,
+  ContractInfo,
+  DAOTrackerContract,
+  ResetDAO as ResetDAOEntity,
+  UControllerOrganization,
+} from '../../types/schema';
+import { createTemplate, equalStrings, fetchTemplateName } from '../../utils';
+
+export function getDAOTrackerContract(address: Address): DAOTrackerContract {
+  let daoTracker = DAOTrackerContract.load(address.toHex()) as DAOTrackerContract;
+  if (daoTracker == null) {
+    daoTracker = new DAOTrackerContract(address.toHex());
+    daoTracker.address = address;
+    let daoTrackerSC = DAOTracker.bind(address);
+    daoTracker.owner = daoTrackerSC.owner();
+    daoTracker.save();
+  }
+  return daoTracker;
+}
+
+export function handleTrackDAO(event: TrackDAO): void {
+  // Ensure the DAOTrackerContract has been added to the store
+  getDAOTrackerContract(event.address);
+
+  let avatar = event.params._avatar;
+  let controller = event.params._controller;
+  let reputation = event.params._reputation;
+  let daoToken = event.params._daoToken;
+  let sender = event.params._sender;
+  let arcVersion = event.params._arcVersion;
+
+  // If the avatar already exists, early out
+  if (AvatarContract.load(avatar.toHex()) != null) {
+    return;
+  }
+
+  // If the sender of the 'track' call is the DaoCreator contract, use its arcVersion
+  let daoCreatorInfo = ContractInfo.load(sender.toHex());
+  if (daoCreatorInfo != null) {
+    if (equalStrings(daoCreatorInfo.name, 'DaoCreator')) {
+      arcVersion = daoCreatorInfo.version;
+    }
+  }
+
+  let avatarTemplate = fetchTemplateName('Avatar', arcVersion);
+  let controllerTemplate = fetchTemplateName('Controller', arcVersion);
+  let reputationTemplate = fetchTemplateName('Reputation', arcVersion);
+  let daoTokenTemplate = fetchTemplateName('DAOToken', arcVersion);
+
+  let missingTemplate = avatarTemplate == null ||
+                        reputationTemplate == null ||
+                        daoTokenTemplate == null;
+
+  let universalController = UControllerOrganization.load(controller.toHex()) != null;
+
+  if (universalController) {
+    missingTemplate = missingTemplate || controllerTemplate == null;
+  }
+
+  if (missingTemplate) {
+    // We're missing a template version in the subgraph
+    return;
+  }
+
+  // Tell the subgraph to start indexing events from the:
+  // Avatar, Controller, DAOToken, and Reputation contracts
+  createTemplate(avatarTemplate, avatar);
+  createTemplate(reputationTemplate, reputation);
+  createTemplate(daoTokenTemplate, daoToken);
+
+  // Track the Controller if it isn't a UController we're already tracking
+  if (universalController === false) {
+    createTemplate(controllerTemplate, controller);
+  }
+
+  // Note, no additional work is needed here because...
+  // * ControllerOrganization is added to the store by the 'RegisterScheme' event
```

Comment:
does the tests cover that ?
indexing a Dao with its schemes ?
how about none universal schemes ? like genesisProrocol ?

Reply:
Yes the test currently adds a ContributionReward scheme and ensures it shows up. Non-universal schemes work as long as the subgraph knows about its address and type in the ContractInfos (at startup).

Reply:
we need to find a way to index NonUni schemes as well without the subgraph knows its address and type in compile time.
we might need additional event for that from daotracker @dOrgJelli